packages/theme/src/cli/commands/theme/share.ts

```diff
@@ -24,12 +26,40 @@ export default class Share extends ThemeCommand {
       description: 'Proceed without confirmation, if current directory does not seem to be theme directory.',
       env: 'SHOPIFY_FLAG_FORCE',
     }),
+    legacy: Flags.boolean({
+      hidden: true,
+      description: 'Use the legacy Ruby implementation for the `shopify theme push` command.',
+      env: 'SHOPIFY_FLAG_LEGACY',
+    }),
   }
 
   static cli2Flags = ['force']
 
   async run(): Promise<void> {
     const {flags} = await this.parse(Share)
+
+    if (flags.password || flags.legacy) {
+      await this.legacyRun()
+      return
+    }
+
+    const pushFlags: PushFlags = {
+      path: flags.path,
+      password: flags.password,
+      store: flags.store,
+      environment: flags.environment,
+      force: flags.force,
+      noColor: flags['no-color'],
+      verbose: flags.verbose,
+      unpublished: true,
```

Comment:
The share command is essentially the push command with `unpublished: true` and the theme is given a random name which is done using `getRandomName`

Reply:
I think we need the following flags (essentially `force` + all `themeFlags` except `environment` )
- `force`
- `store`
- `path`
- `password`

We don't need to pass anything in from `globalFlags`